Peer.UnitTests/Apps/AppBuilderTests.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using CommandLine;
+using Microsoft.Extensions.DependencyInjection;
+using Peer.Apps;
+using Peer.Apps.AppBuilder;
+using Peer.Domain.Configuration;
+using wimm.Secundatives;
+using Xunit;
+
+namespace Peer.UnitTests.Apps;
+
+public class AppBuilderTests
+{
+    private IServiceCollection _services;
+
+    public class WithParseTimeConfig : AppBuilderTests
+    {
+        [Fact]
+        public void ConfigIsCalledWithoutRunningApp()
+        {
+            var called = false;
+            var underTest = Construct()
+                .WithParseTimeServiceConfig(
+                    sc =>
+                    {
+                        called = true;
+                        return new Result<IServiceCollection, ConfigError>(sc.AddSingleton<object>(100));
+                    });
+
+            underTest.WithVerb<Doot>(x => x.WithHandler<DoNothing>());
+            underTest.Build();
+            Assert.True(called);
+        }
+    }
+
+    public class WithSharedServiceConfig : AppBuilderTests
```

Comment:
Given the surrounding contet should this be `WithEuntimeConfig` or something? The shared/global idea seems like it would distinct from config V runtime so over this seems like two of four options; shared config time, shared run time, local config time, local runtime. Not that we need them all to exist before we need them, just looking to confirm my understanding of the existing stuff and the context it exists within.

Reply:
Yeah I think that's a much better way to put it. 

re the states: yeah right now I've been just using it in the handle calls to not change too much (the local runtime config)

Reply:
I'll rename this guy